src/main/java/org/reactome/release/qa/common/QACheckerHelper.java

```diff
@@ -225,5 +232,339 @@ public static String getAttributeTableName(String clsName,
+				physicalEntity.getSchemClass().isa(ReactomeJavaConstants.Polymer) ||
+				physicalEntity.getSchemClass().isa(ReactomeJavaConstants.EntitySet);
+	}
+
+	/**
+	 * Finds all PhysicalEntities contained within a Complex, Polymer or EntitySet. Searches recursively by calling parent 'findAllPhysicalEntities' method.
+	 * @param multiPEInstance GKInstance -- Complex, Polymer or EntitySet instance that will be searched for all PhysicalEntity instances.
+	 * @return Set<GKInstance> -- All distinct PhysicalEntities that are found in Complex/Polymer/EntitySet.
+	 * @throws Exception -- Thrown by MySQLAdaptor.
+	 */
+	public static Set<GKInstance> findAllConstituentPEs(GKInstance multiPEInstance) throws Exception {
+		Set<GKInstance> physicalEntities = new HashSet<>();
+		if (multiPEInstance.getSchemClass().isa(ReactomeJavaConstants.Complex) || multiPEInstance.getSchemClass().isa(ReactomeJavaConstants.Polymer)) {
+			// Recursively search Complex/Polymer for all PhysicalEntities.
+			for (GKInstance constituentPE : getComplexOrPolymerConstituentPEs(multiPEInstance)) {
+				physicalEntities.addAll(findAllPhysicalEntities(constituentPE));
+			}
+		} else if (multiPEInstance.getSchemClass().isa(ReactomeJavaConstants.EntitySet)) {
+			// Recursively search members and candidates (if EntitySet is a CandidateSet) for all PhysicalEntities.
+			physicalEntities.addAll(findEntitySetPhysicalEntities(multiPEInstance));
+		}
+		return physicalEntities;
+	}
+
+	/**
+	 * Helper method that returns either a Complexes' components or Polymers' repeatedUnits.
+	 * @param multiPEInstance GKInstance -- Either a Complex or Polymer instance.
+	 * @return Set<GKInstance> -- All components or repeatedUnits from the incoming instance.
+	 * @throws Exception -- Thrown by MySQLAdaptor.
+	 */
+	private static Collection<GKInstance> getComplexOrPolymerConstituentPEs(GKInstance multiPEInstance) throws Exception {
+		Set<GKInstance> constituentPEs = new HashSet<>();
+		if (multiPEInstance.getSchemClass().isa(ReactomeJavaConstants.Complex)) {
+			constituentPEs.addAll(multiPEInstance.getAttributeValuesList(ReactomeJavaConstants.hasComponent));
+		} else if (multiPEInstance.getSchemClass().isa(ReactomeJavaConstants.Polymer)) {
+			constituentPEs.addAll(multiPEInstance.getAttributeValuesList(ReactomeJavaConstants.repeatedUnit));
+		}
+		return constituentPEs;
+	}
+
+	/**
+	 * Finds all PhysicalEntities that exist within an EntitySet by searching for all member PEs (via 'hasMember' attribute)
+	 * and all candidate PEs (if entitySet is a CandidateSet; via 'hasCandidate attribute).
+	 * @param entitySet GKInstance -- EntitySet instance that will be searched for all PhysicalEntities.
+	 * @return Set<GKInstance> -- All distinct PhysicalEntities that are found in EntitySet.
+	 * @throws Exception -- Thrown by MySQLAdaptor.
+	 */
+	public static Set<GKInstance> findEntitySetPhysicalEntities(GKInstance entitySet) throws Exception {
```

Comment:
Please consider using static methods in InstanceUtilities to avoid any duplication.